frontend/src/lib/components/LemonTable/LemonTable.tsx

```diff
@@ -0,0 +1,80 @@
+import clsx from 'clsx'
+import React, { HTMLProps } from 'react'
+import './LemonTable.scss'
+
+export interface LemonTableColumn<T extends Record<string, any>, D extends keyof T> {
+    title?: string | React.ReactNode
+    key?: keyof T
+    dataIndex?: D
+    render?: (dataValue: T[D] | undefined, record: T) => React.ReactNode
+    sorter?: (a: T, b: T) => number
+    span?: number
+    className?: string
+    /** Column content alignment. Useful e.g. for columns showing numbers, which should be right-aligned */
+    align?: 'left' | 'right'
+    /** TODO: Whether the column should be sticky when scrolling */
+    sticky?: boolean
+    /** TODO: Set width */
+    width?: string | number
+}
+
+export type LemonTableColumns<T extends Record<string, any>> = LemonTableColumn<T, keyof T>[]
+
+export interface LemonTableProps<T extends Record<string, any>> {
+    columns: LemonTableColumns<T>
+    dataSource: T[]
+    rowKey?: string
+    onRow?: (record: T) => Omit<HTMLProps<HTMLTableRowElement>, 'key'>
+    'data-attr'?: string
+}
+
+function isHorizontallyScrollable(element: HTMLElement): boolean {
+    return element.scrollWidth > element.clientWidth
+}
+
+export function LemonTable<T extends Record<string, any>>({
+    columns,
+    dataSource,
+    rowKey,
+    onRow,
+    ...divProps
+}: LemonTableProps<T>): JSX.Element {
+    return (
+        <div className="LemonTable" {...divProps}>
+            <table>
+                <thead>
+                    <tr>
+                        {columns.map((headerCol, headerColIndex) => (
+                            <th
+                                key={headerColIndex}
+                                className={clsx(headerCol.sticky && 'LemonTable__cell--sticky', headerCol.className)}
+                                align={headerCol.align}
+                            >
+                                {headerCol.title}
+                            </th>
+                        ))}
+                    </tr>
+                </thead>
+                <tbody>
+                    {dataSource.map((data, rowIndex) => (
```

Comment:
The after table is looking so much better. Nice job! One of the biggest problems with our older tables was that it would lag with many rows (i.e., events table). I don't know if you have this mind already, but we could probably make leaps in performance with some kind of virtualization here. What do you think?

Reply:
Definitely something to consider, will try some virtualization out in events/persons when I get to them!